docs/dev_install.rst

```diff
@@ -34,7 +34,7 @@ Installing Depedencies
     mkdir build
     cd build
     cmake ..
-    make -j 8
+    make -j $(nproc 2>/dev/null || sysctl -n hw.logicalcpu 2>/dev/null || echo 8)
```

Comment:
Thanks for updating this! I actually think `make -j 8` is better, given the newer version is too scary. 
Note that having threads more than #CPU cores isn't really a disaster, except for out-of-memory of cache contention - which are unlikely to happen. Even if those happens, I believe most users know that they should simply reduce the number `8`.

Reply:
Thanks for your patience, 

lol its too lengthy indeed, but here is what our company does which I find a bit helpful: write a ./Allmake script to handle all these lengthy input flags/variables at higher level, and also treats differently depending on the operating system. So what people have to do is simply `` chmod +x Allmake `` then `` ./Allmake``

An example:
```
if [ "$(uname)" == "Darwin" ]; then
    echo "Mac is currectly not supported"       
elif [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]; then
    echo "Detected plattform: Linux not Mac"
    echo "Compiling Debug mode"
    echo "Compiling Release mode"
elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW64_NT" ] || [ "$(expr substr $(uname -s) 1 10)" == "MINGW32_NT" ]; then
    echo "Detected plattform: Win"
    echo "Compiling Debug mode"
    echo "Compiling Release mode"
fi
```

We can create later if needed, but for now I will just reset to `` -j 8 ``